### ISSUE TITLE
Not updating to cwd when using an autocmd

### ISSUE THREAD

**Issue by willswats** [kyazdani42/nvim-tree.lua#949]
**OS**
Fedora 35

**Neovim version**
NVIM v0.6.1

**Nvim-tree version**
Master

**Describe the bug**
When using an `autocmd` to update the current working directory to the directory of the current buffer, nvim-tree is not updated. 

The `autocmd` used:

```
autocmd BufEnter * silent! lcd %:p:h
```

**To Reproduce**
1. Setup this `autocmd`:

```
autocmd BufEnter * silent! lcd %:p:h
```

2. Navigate to a file in nvim-tree and open it.

**Expected behavior**
For nvim-tree to be updated to the current working directory.

**Comment by kyazdani42** [comment]
can you try adding `+nested` to your autocmd ?

**Comment by willswats** [comment]
Where should I add it? I've tried adding `+nested` in these two locations, but it's had no effect:

```
autocmd BufEnter * silent! lcd %:p:h +nested
autocmd BufEnter * silent! +nested lcd %:p:h
```

**Comment by kyazdani42** [comment]
`:help autocmd-nested`

**Comment by willswats** [comment]
I see it's `++nested` in the help, and so I've added it here, but it's had no effect as well:

```
 autocmd BufEnter * ++nested silent! lcd %:p:h
```

**Comment by kyazdani42** [comment]
is `update_cwd` true in the setup call for nvim-tree ?

**Comment by willswats** [comment]
Yes it is, I have my setup for nvim-tree [here](https://github.com/willswats/neovim-config/blob/main/lua/user/nvim-tree.lua) if that helps. Also if I open a file from my desktop the `autocmd` works and nvim-tree is updated, it only doesn't work when I open Neovim and then navigate to a file.

**Comment by willswats** [comment]
It works, thanks a lot for the help.
